src/prototype.rs

```diff
@@ -82,16 +84,18 @@ struct FirstPass<'a> {
     text: &'a str,
     tree: Tree<Item>,
     last_line_blank: bool,
+    references: Vec<RefDefScan<'a>>,
```

Comment:
I think this is a performance concern as the number of definitions grows. I think the win is to use [unicase](https://crates.io/crates/unicase) rather than caseless, so that the wrapped string can be used as a key in either a HashMap or a BTreeMap.